### PR TITLE
[jsonrpc refactoring 1/n] - reorg and consolidate rpc api modules

### PR DESCRIPTION
**Pull request by patrickkuo**
Reorg and consolidate jsonrpc modules before more major refactoring/changes. Some of the module segregation was remnant of Gateway deprecation.

### CODE REVIEW

File: crates/sui-json-rpc/src/api/bls.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
```
Comment (gegaowp) [inline crates/sui-json-rpc/src/api/bls.rs:1]: to double check, files under api/ are just split results of api.rs right?

File: crates/sui-open-rpc/spec/openrpc.json [hunk 2]
```diff
@@ -3448,39 +3449,34 @@
...
-              "bcs_bytes",
               "dataType",
+              "fields",
               "has_public_transfer",
-              "type",
-              "version"
+              "type"
             ],
             "properties": {
-              "bcs_bytes": {
```
Comment (gegaowp) [inline crates/sui-open-rpc/spec/openrpc.json:3458]: supposedly this PR should not break the "schema", why do we have generated changes here?

**Review by gegaowp — COMMENTED** [review]
LG overall!

File: crates/sui-json-rpc/src/api/bls.rs (hunk 1 above)
Comment (patrickkuo) [reply crates/sui-json-rpc/src/api/bls.rs:1]: Yes, this is to make it easier to maintain

File: crates/sui-open-rpc/spec/openrpc.json (hunk 2 above)
Comment (patrickkuo) [reply crates/sui-open-rpc/spec/openrpc.json:3458]: Nicely spotted lol, there was a schema clash between Raw and parsed data, the old schema wrongly picked up the “RawData” schema. This is due to the generic data structure, I will change them to enum in next PR

**Review by gegaowp — APPROVED** [review]
let's ship it!
wee-ship

**Pull request merged**